OrmTester/OrmTesterDesktop/Utils/ChartGenerationHelper.cs

```diff
@@ -191,18 +191,25 @@ private IEnumerable<StatisticParameter> GetParamsByOperation(OperationType opera
 
         private Tuple<double, double, double, double> GetAverageForFrameworks(IEnumerable<StatisticParameter> statisticParameters)
         {
-            if (statisticParameters.Any())
-            {
-                var nHibernateCreateAverageBulk = statisticParameters.Where(param => param.IsBulk).Average(createParameter => createParameter.NHibernateExecutionTimePerRecord);
-                var efCreateAverageBulk = statisticParameters.Where(param => param.IsBulk).Average(createParameter => createParameter.EfExecutionTimePerRecord);
-                var nHibernateCreateAverageSingle = statisticParameters.Where(param => !param.IsBulk).Average(createParameter => createParameter.NHibernateExecutionTimePerRecord);
-                var efCreateAverageSingle = statisticParameters.Where(param => !param.IsBulk).Average(createParameter => createParameter.EfExecutionTimePerRecord);
-                return new Tuple<double, double, double, double>(nHibernateCreateAverageBulk, efCreateAverageBulk, nHibernateCreateAverageSingle, efCreateAverageSingle);
+            var bulkResults = statisticParameters.Where(param => param.IsBulk);
+            var nHibernateCreateAverageBulk = 0.0;
+            var efCreateAverageBulk = 0.0;
+            if (bulkResults.Any())
+            {
+                nHibernateCreateAverageBulk = bulkResults.FirstOrDefault().NHibernateExecutionTimePerRecord;
+                efCreateAverageBulk = bulkResults.FirstOrDefault().EfExecutionTimePerRecord;                
+                
             }
-            else
+            var singleResutls = statisticParameters.Where(param => !param.IsBulk);
+            var nHibernateCreateAverageSingle = 0.0;
+            var efCreateAverageSingle = 0.0;
+            if (singleResutls.Any())
             {
-                return null;
+                nHibernateCreateAverageSingle = singleResutls.FirstOrDefault().NHibernateAverage;
```

Comment:
może warto użyć ?. jakby kolekcja była pusta to się wywali

Reply:
ale to trzeba będzie w połowie klasy przekonwertować z double na double?

Reply:
racja to może jakiś ifik na początku sprawdzający czy coś w tej kolekcji wgl jest, może zwracać default(double)

Reply:
no to jest if wyżej